lib/matplotlib/backends/backend_pgf.py

```diff
@@ -76,33 +76,30 @@ def get_preamble():
 
 NO_ESCAPE = r"(?<!\\)(?:\\\\)*"
 re_mathsep = re.compile(NO_ESCAPE + r"\$")
-re_escapetext = re.compile(NO_ESCAPE + "([_^$%])")
-repl_escapetext = lambda m: "\\" + m.group(1)
-re_mathdefault = re.compile(NO_ESCAPE + r"(\\mathdefault)")
-repl_mathdefault = lambda m: m.group(0)[:-len(m.group(1))]
+_replace_escapetext = functools.partial(
+    re.compile(NO_ESCAPE + "(?=[_^$%])").sub, lambda m: "\\")
+_replace_mathdefault = functools.partial(
+    re.compile(NO_ESCAPE + r"(\\mathdefault)").sub, lambda m: "")
```

Comment:
I think these regex substitutions are simpler to read: they simply mean, in the first case: "when the next character is a _, ^, $ or %" (not preceded by an escape), insert a backslash" and in the second case: "replace \mathdefault (not preceded by an escape) by the empty string".

Reply:
I suggest you take this opportunity to put your English translations in as comments.  The old and new versions are incomprehensible to me, and I suspect to many others, without extensive research.  I have no objection to your new versions, though.

Reply:
done